src/nupic/regions/ScalarSensor.cpp

```diff
@@ -66,15 +64,20 @@ ScalarSensor::ScalarSensor(BundleIO &bundle, Region *region)
   deserialize(bundle);
 }
 
+  void ScalarSensor::compute()
+  {
+    Real32* array = (Real32*)encodedOutput_->getData().getBuffer();
+    UInt uintArray[encoder_->getOutputWidth()];
+    const Int32 iBucket = encoder_->encodeIntoArray(sensedValue_, uintArray);
+    ((Int32*)bucketOutput_->getData().getBuffer())[0] = iBucket;
+    for(UInt i=0; i<encoder_->getOutputWidth(); i++) //FIXME optimize
+    {
+      array[i] = (Real32)uintArray[i]; // copy values back to SP's 'array' array
+    }
+  }
 
 ScalarSensor::~ScalarSensor() { delete encoder_; }
 
-void ScalarSensor::compute() {
-  Real32 *array = (Real32 *)encodedOutput_->getData().getBuffer();
-  const Int32 iBucket = encoder_->encodeIntoArray(sensedValue_, array);
-  ((Int32 *)bucketOutput_->getData().getBuffer())[0] = iBucket;
-}
-
 /* static */ Spec *ScalarSensor::createSpec() {
```

Comment:
You also need to set the type of the output in the Output Spec:
```
  ns->outputs.add("encoded", OutputSpec("Encoded value", NTA_BasicType_Real32,
                                        0,    // elementCount
                                        true, // isRegionLevel
                                        true  // isDefaultOutput
                                        ));
```
The NTA_BasicType_Real32 should be NTA_BasicType_UInt32

Now, the output of the ScalerSensor is normally an SP region which is currently implemented in Python.  It's input Spec is probably expecting a Real32.  This might not matter since Python allows you to be sloppy with data types but you might want to verify that this  is going to work.

Reply:
Thanks for spotting that, I didn't know this!